File: rust/src/model/domain/account.rs
pub const INVENTORY_ACCOUNT: &str = "10.01";
pub const INVENTORY_DIFF_ACCOUNT: &str = "10.02";
pub const RECEIVABLE_ACCOUNT: &str = "62.01";
pub const PAYABLE_ACCOUNT: &str = "60.01";
pub const SALES_ACCOUNT: &str = "90.01";
pub const COGS_ACCOUNT: &str = "90.02";

pub const DEBT: &str = "D";
pub const CREDIT: &str = "C";

#[derive(Debug, Clone)]
pub struct Account{
    pub id: String,
    pub name: String,
}
impl Account {
    pub fn new(aid: &str, aname: &str) -> Account {
        Account { id: aid.to_string(), name: aname.to_string() }
    }

    pub fn chart_of_accounts() -> Vec<Account> {
        vec![
            Account::new(INVENTORY_ACCOUNT, "Inventory"),
            Account::new(INVENTORY_DIFF_ACCOUNT, "Inventory differences"),
            Account::new(RECEIVABLE_ACCOUNT, "Account receivable"),
            Account::new(PAYABLE_ACCOUNT, "Account payable"),
            Account::new(SALES_ACCOUNT, "Sales"),
            Account::new(COGS_ACCOUNT, "Cost of goods sold"),
        ]
    }
}

File: golang/model/domain/account.go
package domain

const INVENTORY_ACCOUNT string = "10.01"

const INVENTORY_DIFF_ACCOUNT string = "10.02"

const RECEIVABLE_ACCOUNT string = "62.01"

const PAYABLE_ACCOUNT string = "60.01"

const SALES_ACCOUNT string = "90.01"

const COGS_ACCOUNT string = "90.02"

const DEBT string = "D"

const CREDIT string = "C"

type Account struct {
	Id   string
	Name string
}

func NewAccount(id, name string) Account {
	return Account{id, name}
}

func ChartOfAccounts() []Account {
	return []Account{
		NewAccount(INVENTORY_ACCOUNT, "Inventory"),
		NewAccount(INVENTORY_DIFF_ACCOUNT, "Inventory differences"),
		NewAccount(RECEIVABLE_ACCOUNT, "Account receivable"),
		NewAccount(PAYABLE_ACCOUNT, "Account payable"),
		NewAccount(SALES_ACCOUNT, "Sales"),
		NewAccount(COGS_ACCOUNT, "Cost of goods sold"),
	}
}
